[rtl] Implement mvendorid/marchid/mimpid CSRs (#1338)
lowRISC/ibex: doc/02_user/getting_started.rst

Ibex comes with three different register file implementations that can be selected using the enumerated parameter ``RegFile`` defined in :file:`rtl/ibex_pkg.sv`.
Depending on the target technology, either the flip-flop-based ("ibex_pkg::RegFileFF", default), the latch-based ("ibex_pkg::RegFileLatch") or an FPGA-targeted ("ibex_pkg::RegFileFPGA") implementation should be selected.
For more information about the three register file implementations and their trade-offs, check out :ref:`register-file`.

Identification CSRs
-------------------

> The RISC-V Privileged Architecture specifies several read-only CSRs that can be to identify the vendor and microarchitecture of a CPU.

Reviewer: ```suggestion
The RISC-V Privileged Architecture specifies several read-only CSRs that identify the vendor and micro-architecture of a CPU.
```

micro-architecture for consistency with the usage below.

These are ``mvendorid``, ``marchid`` and ``mimpid``.
The fixed, read-only values for these CSRs are defined in :file:`rtl/ibex_pkg.sv`.
Implementers should carefully consider appropriate values for these registers.

> Ibex, as an open source implementation, has an assigned architecture ID (``marchid``) of 22 (Allocations are specified in `marchid.md of the riscv-isa-manual repository <https://github.com/riscv/riscv-isa-manual/blob/master/marchid.md>`_).

Reviewer: ```suggestion
Ibex, as an open source implementation, has an assigned architecture ID (``marchid``) of 22. 
(Allocations are specified in `marchid.md of the riscv-isa-manual repository <https://github.com/riscv/riscv-isa-manual/blob/master/marchid.md>`_.)
```
